business/services.go

```diff
@@ -53,54 +60,66 @@ func (in *SvcService) GetServiceList(namespace string, linkIstioResources bool)
 	go func() {
 		defer wg.Done()
 		var err2 error
+		var selectorLabels map[string]string
+		if criteria.ServiceSelector != "" {
+			if selector, err3 := labels.ConvertSelectorToLabelsMap(criteria.ServiceSelector); err3 == nil {
+				selectorLabels = selector
+			} else {
+				log.Warningf("Services not filtered. Selector %s not valid", criteria.ServiceSelector)
+			}
+		}
 		// Check if namespace is cached
 		// Namespace access is checked in the upper call
-		if IsNamespaceCached(namespace) {
-			svcs, err2 = kialiCache.GetServices(namespace, nil)
+		if IsNamespaceCached(criteria.Namespace) {
+			svcs, err2 = kialiCache.GetServices(criteria.Namespace, selectorLabels)
 		} else {
-			svcs, err2 = in.k8s.GetServices(namespace, nil)
+			svcs, err2 = in.k8s.GetServices(criteria.Namespace, selectorLabels)
```

Comment:
Does it still stay with kialiCache and k8s API calls within this refactoring, or another PR will come to fetch information from Registry?

Reply:
Yes, the information registry will come later.
This is a first step to make the APIs more clear and to redirect the use of the services (and workloads) to the proper business/{services,workload,app,etc}.go file.

Once that API is refactored, the services are fetched from a single point and changes like using registry and/or other sources would be more transparent for the other layers.